cluster-operator/src/main/java/io/strimzi/operator/cluster/Main.java

```diff
@@ -105,7 +107,14 @@ public static void main(String[] args) {
     static CompositeFuture run(Vertx vertx, KubernetesClient client, PlatformFeaturesAvailability pfa, ClusterOperatorConfig config) {
         Util.printEnvInfo();
 
-        ResourceOperatorSupplier resourceOperatorSupplier = new ResourceOperatorSupplier(vertx, client, pfa, config.getOperationTimeoutMs());
+        ResourceOperatorSupplier resourceOperatorSupplier = new ResourceOperatorSupplier(
+                vertx,
+                client,
+                pfa,
+
+                config.getOperationTimeoutMs(),
+                KubernetesRestartEventPublisher.createPublisher(client, config.getOperatorName(), pfa.hasEventsApiV1())
```

Comment:
Why do we pass this to the ResourceOperatorSupplier from the outside instead of creating it inside of it?

Reply:
Fixed.